lib/shopping_cart.rb

```diff
@@ -0,0 +1,49 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products, :details, :is_full
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.delete_suffix("items").to_i
+    @products = []
+    @details = {}
+    @details[:name] = name
+    @details[:capacity] = @capacity
```

Comment:
this is a lot to put in initialize.. usually you want to steer away from formatting in this method. I would make a details method that has this in it.